src/Pelagos/Entity/DatasetSubmission.php

```diff
@@ -1129,16 +1139,13 @@ public function validate(ExecutionContextInterface $context)
             }
         }
 
-        $coldStorageViolationMsg = 'You must provide both File Size and Sha256 Hash value for Cold Storage Information.';
-        if (null !== $this->datasetFileColdStorageArchiveSize && null === $this->datasetFileColdStorageArchiveSha256Hash) {
-                $context->buildViolation($coldStorageViolationMsg)
-                    ->atPath('datasetFileColdStorageArchiveSha256Hash')
-                    ->addViolation();
-        }
-
-        if (null === $this->datasetFileColdStorageArchiveSize && null !== $this->datasetFileColdStorageArchiveSha256Hash) {
+        $coldStorageViolationMsg = 'You must provide file size, sha256 hash, and original '
+        . 'filename for Cold Storage Information.';
+        if (false === (PelagosUtilities::nullOrNone(array($this->this->datasetFileColdStorageArchiveSize,
+            $this->datasetFileColdStorageArchiveSha256Hash,
+            $this->datasetFileColdStorageOriginalFilename)))) {
```

Comment:
How would it get set in the first place? I'm not sure whether I totally understand

Reply:
It would be set normally from either the review form, or via the bulk-loader command. This just enforced the "all 3 or none at all" rule for the form.

Reply:
Removed this as it is no longer needed with a public method that sets all 3 attributes and once and does not allow nulls, in place of the 3 separate public setters.